portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -98,11 +109,29 @@ private static String getParameter(HttpServletRequest request, String name, Stri
     return value;
   }
 
-  private static void storeComments(String name, String email, String comment, long timestamp) {
+  private static Optional<Double> getSentiment(String message) {
+    try {
+      Document doc =
+          Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+      LanguageServiceClient languageService = LanguageServiceClient.create();
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+      double score = sentiment.getScore();
+      languageService.close();
+      return Optional.of(score);
+    } catch (IOException e) {
+      e.printStackTrace();
+      return Optional.empty();
+    }
+  }
+
+  private static void storeComments(String name, String email, String comment, Optional<Double> sentiment, long timestamp) {
     Entity taskEntity = new Entity(ENTITY_KIND);
     taskEntity.setProperty(ENTITY_NAME_HEADER, name);
     taskEntity.setProperty(ENTITY_EMAIL_HEADER, email);
     taskEntity.setProperty(ENTITY_COMMENT_HEADER, comment);
+    if (sentiment.isPresent()) {
```

Comment:
You can use sentimentOptional.ifPresent(sentiment -> taskEntity.setProperty(ENTITY_SENTIMENT_HEADER, sentiment))

Reply:
Done